# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/linux26/arch-ppc-smp-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

This is only a quick fix to make the 2.6 kernel compile on PowerPC.
Untested due to the lack of SMP hardware ...

  - Rene Rebe <rene@rocklinux.org>

--- linux-2.6.0-test4/arch/ppc/kernel/smp.c.orig	2003-08-23 02:00:45.000000000 +0200
+++ linux-2.6.0-test4/arch/ppc/kernel/smp.c	2003-09-06 00:56:46.000000000 +0200
@@ -47,7 +47,6 @@
 DEFINE_PER_CPU(unsigned int, prof_multiplier);
 DEFINE_PER_CPU(unsigned int, prof_counter);
 unsigned long cache_decay_ticks = HZ/100;
-unsigned long cpu_online_map = cpumask_of_cpu(0);
 unsigned long cpu_possible_map = 1UL;
 int smp_hw_index[NR_CPUS];
 struct thread_info *secondary_ti;
@@ -361,6 +360,7 @@
 
 void __devinit smp_prepare_boot_cpu(void)
 {
+	unsigned long cpu_online_map = cpumask_of_cpu(0);
 	cpu_set(smp_processor_id(), cpu_online_map);
 	cpu_set(smp_processor_id(), cpu_possible_map);
 }
@@ -399,6 +399,7 @@
 	struct task_struct *p;
 	char buf[32];
 	int c;
+	unsigned long cpu_online_map = cpumask_of_cpu(0);
 
 	/* create a process for the processor */
 	/* only regs.msr is actually used, and 0 is OK for it */
